### ISSUE TITLE
Multispeaker training ends up with voices tending toward average

### ISSUE THREAD

**Issue by iamanigeeit** [espnet/espnet#5301]
This is in relation to https://github.com/espnet/espnet/discussions/5253

I am trying to train a multispeaker FS2 model for LJSpeech + [ESD](https://github.com/HLTSingapore/Emotional-Speech-Data) to add variability in speech. My generated samples for the ESD speakers ended up sounding like the LJSpeech speaker in voice quality with some pitch difference -- in a listening test, they would never be considered the same speaker.

I changed from using speaker ids to X-vectors in concat mode, but the result is still the same; in fact the output is not distinguishable.

I thought this was a data problem because there is 20x more data per speaker in LJSpeech vs ESD. So i removed all LJSpeech samples from the training data and changed the learning rate to 5x, then continued training on ESD only. This made very little difference.

But i realized **the [pretrained VCTK models](https://drive.google.com/drive/folders/1Zv1HIPUEGKcBbGHF_ay6YIHtgp5CLYcv) also have this averaged-voice problem**. I think the GST + X-vector + Tacotron2 samples sound somewhat like the original speaker, but all the rest don't.

It seems that the multispeaker models that work -- [DeepVoice3](https://r9y9.github.io/deepvoice3_pytorch/) and [YourTTS](https://edresson.github.io/YourTTS/) **inject the speaker embedding in the decoder**. This makes sense because our current method (adding to encoder `hs` / concat + project) lets the speaker information be "lost" in the encoded speech representation. In fact, in YourTTS, the speaker embedding does not interact with `hs` at all. It goes to the duration predictor (because different speakers generate different phone durations), then to the decoder and vocoder (because applying speaker characteristics can be seen as output audio transformation).

I'll see if i can do this for FS2 next month. If anyone has ideas, please post below!

**Comment by mindmapper15** [comment]
Can you post a couple of voice samples from TTS?

**Comment by iamanigeeit** [comment]
Here's the one using sids after 700+ epochs:

[Originals.zip](https://github.com/espnet/espnet/files/12048503/Originals.zip)
[Generated.zip](https://github.com/espnet/espnet/files/12048508/Generated.zip)

**Comment by iamanigeeit** [comment]
Interestingly, using a reference encoder for prosody transfer also causes entanglement with text and speaker information. The Daft-Exprt paper (https://arxiv.org/pdf/2108.02271.pdf) tries to solve it by injecting the prosody vector as parameters in new layers in the encoder, variance predictor, and decoder.

**Comment by mindmapper15** [comment]
> Here's the one using sids after 700+ epochs:
> 
> [Originals.zip](https://github.com/espnet/espnet/files/12048503/Originals.zip) [Generated.zip](https://github.com/espnet/espnet/files/12048508/Generated.zip)

What X-vector did you used during inference?
Did you used one single X-vector for all synthesis? or different X-vectors for each synthesis?

The prosody elements in FastSpeech 2 are determined from text + x-vector embedding.
If you use same text and same x-vector for all synthesis, naturally every synthesized audio sound like exactly same because every input data, which is concat (or addition) of text encoder output and xvector, from encoder is all exactly same.

VITS TTS introduced some random element at duration predictor and decoder, though. Maybe you should checkout that model.

**Comment by iamanigeeit** [comment]
I followed `tts.sh` for x-vectors, that is, extracting the x-vector for every utterance during training. And during inference i used the speaker-averaged x-vector. Using different x-vectors make them sound slightly different, but they sound averaged, for example the male voices are higher and the female voices are lower than they should be.

The pretrained models [here](https://drive.google.com/drive/folders/1Zv1HIPUEGKcBbGHF_ay6YIHtgp5CLYcv) also have this problem.

I still want to use FS2, because it's easy to train overnight, then look at the results and try something else the next day.
